controllers/nodemaintenance_controller.go

```diff
@@ -116,15 +119,16 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		r.logger.Info("Error reading the request object, requeuing.")
 		return emptyResult, err
 	}
-
-	// Add finalizer when object is created
 	drainer, err := createDrainer(ctx, r.MgrConfig)
 	if err != nil {
 		return emptyResult, err
+		// Add finalizer when object is created
 	}
 
 	if !controllerutil.ContainsFinalizer(nm, v1beta1.NodeMaintenanceFinalizer) && nm.ObjectMeta.DeletionTimestamp.IsZero() {
 		controllerutil.AddFinalizer(nm, v1beta1.NodeMaintenanceFinalizer)
+		// begin maintenance on adding finalizer
+		utils.NormalEvent(r.Recorder, nm, utils.EventReasonBeginMaintenance, utils.EventMessageBeginMaintenance)
```

Comment:
Nit: I think we should omit the event only after the finalizer is added (i.e after Update was successful)

Reply:
I thought the opposite, but I see your motivation